designs/2023-relative-cache/README.md

```diff
@@ -0,0 +1,65 @@
+- Repo: eslint/eslint
+- Start Date: 2023-02-14
+- RFC PR: <https://github.com/eslint/rfcs/pull/106>
+- Authors: Christian Schulz (@cschulz)
+
+# Relative cache location strategy
+
+## Summary
+
+To be able to share the created eslint cache between different systems f.e. local machine and CI, there is a need for a relative path stored in the cache.
+
+## Motivation
+
+The overall motivation for doing this is performance during the eslint process.
+
+Most people are running eslint on local developer machine and additional on the CI workers.
+In high changing repositories there are several pull requests at the same time changing different files requiring linting all the time.
+This is a time consuming process depending on the enabled rules like type checking rules.
+
+If the CI worker is able to cache its last run, it can just pick this cache up and use it.
+Also developers can reuse this cache or commit it as part of a repository for the CI.
+
+## Detailed Design
+
+The LintResultCache takes a file path as parameter to find or store a file cache entry.
```

Comment:
We need more details in this section. It should be written in such a way that someone else would be able to implement it. Please reference filenames, classes, etc., and what would have to change so we can get an idea of the implementation complexity. We'd like your recommendation for the best way to implement your suggestion. As-is, there's not enough detail here to comment on.

Reply:
@nzakas Can you help me out here? I am not so familar with the code to answer such questions in this detail level.

Reply:
@CSchulz I completely understand. Part of the RFC process is to have you get familiar with the code to see how plausible it is to implement (otherwise, we will have to do that for every RFC and there's just not enough time for that).

For this change, start by looking in `lib/eslint/flat-eslint.js`.